NumLib/Extrapolation/LocalLinearLeastSquaresExtrapolator.cpp

```diff
@@ -70,12 +70,12 @@ void LocalLinearLeastSquaresExtrapolator::extrapolate(
 
     if (!_nodal_values ||
 #ifdef USE_PETSC
-        static_cast<std::size_t>(_nodal_values->getLocalSize() +
+        static_cast<int>(_nodal_values->getLocalSize() +
                                  _nodal_values->getGhostSize())
 #else
         _nodal_values->size()
 #endif
-            != num_nodal_dof_result)
+            != static_cast<int>(num_nodal_dof_result))
```

Comment:
Not sure about this casts. The sizes of a GlobalVector are probably the largest integers available, so the std::size_t used before here seems better sized.
I'd continue to use std::size_t here unless there are some other arguments to use an int.

Reply:
Using GlobalIndexType in one of the casts solves the issue. The change is already uploaded.